pkg/action/executor/k8s_controller.go

```diff
@@ -32,6 +33,7 @@ import (
 type k8sController interface {
 	get(name string) (*k8sControllerSpec, error)
 	update(updatedSpec *k8sControllerSpec) error
+	waitForReplicaSetToBeScheduled(name string) (bool, error)
```

Comment:
Why do we need to define this function in the interface?  I think `waitForReplicaSetToBeScheduled` should be a private function for `parentController`.